marklogic-data-hub-spark-connector/src/main/java/com/marklogic/hub/spark/sql/sources/v2/DefaultSource.java

```diff
@@ -235,30 +244,51 @@ private DocumentMetadataHandle buildDocumentMetadata() {
     }
 
     private String initializeJob(StructType schema) {
-        loadJobEndpoints();
-
         ObjectNode sparkMetadata = objectMapper.createObjectNode();
         try {
             sparkMetadata.set("schema", objectMapper.readTree(schema.json()));
         } catch (Exception e) {
             logger.warn("Unable to read Spark schema as a JSON object; cause: " + e.getMessage() + "; the schema will not " +
                 "be persisted on the job document");
         }
-        String jobId = SparkService.on(hubClient.getJobsClient()).initializeJob(sparkMetadata);
+
+        String jobId = SparkService.on(hubClient.getJobsClient(), customJobApiDefinitions.getInitializeJobApiDefinition()).initializeJob(sparkMetadata);
         logger.info("Initialized job; ID: " + jobId);
         return jobId;
     }
 
-    private void loadJobEndpoints() {
-        String initializeJobApiPath = "/marklogic-data-hub-spark-connector/initializeJob.api";
-        String initializaJobScriptPath = "/marklogic-data-hub-spark-connector/initializeJob.sjs";
-        loadModuleIfNotPresent(initializaJobScriptPath, Format.TEXT);
-        loadModuleIfNotPresent(initializeJobApiPath, Format.JSON);
+    private CustomJobApiDefinitions readCustomJobApiDefinitions(Map<String, String> options) {
+        InputStreamHandle initializeDefinition = null;
+        String key = "initializejobapipath";
```

Comment:
Shouldn't this be "custominitializejobapipath"?

Reply:
No need for "custom" - it's apparent that it's custom since the user is including it. We don't use "custom" as a prefix for "ingestendpointparams" either.